app/forms/creator_settings_form.rb

```diff
@@ -13,16 +13,14 @@ class CreatorSettingsForm
   attr_accessor :community_name, :primary_brand_color_hex, :invite_only_mode, :public, :logo,
                 :checked_code_of_conduct, :checked_terms_and_conditions
 
-  # TODO: determine the default values
-  def initialize(community_name: nil, primary_brand_color_hex: nil, invite_only_mode: nil, public: nil,
-                 checked_code_of_conduct: false, checked_terms_and_conditions: false, logo: nil)
-    @community_name = community_name
-    @primary_brand_color_hex = primary_brand_color_hex
-    @invite_only_mode = invite_only_mode
-    @public = public
-    @logo = logo
-    @checked_code_of_conduct = checked_code_of_conduct
-    @checked_terms_and_conditions = checked_terms_and_conditions
+  def initialize(attributes = {})
+    super
+
+    @primary_brand_color_hex = primary_brand_color_hex || ::Settings::UserExperience.primary_brand_color_hex
+    @invite_only_mode = invite_only_mode || 1
+    @public = public || 1
```

Comment:
The `public` here clearly confuses the syntax highlighter but it should work ok as Ruby allows for overriding these names (though it may still be confusing for the reader).